src/redux/books/booksSlice.js

```diff
@@ -0,0 +1,29 @@
+const ADDBOOK = 'bookStore/books/ADDBOOK';
+const REMOVEBOOK = 'bookStore/books/REMOVEBOOK';
+
+const initialState = [];
+
+export const addBook = (payload) => ({
+  type: ADDBOOK,
+  payload,
+});
+
+export const removeBook = (id) => ({
+  type: REMOVEBOOK,
+  id,
+});
+
+const reducer = (state = initialState, action) => {
+  switch (action.type) {
+    case ADDBOOK:
+      return [...state, {
+        ...action.payload,
+      }];
+    case REMOVEBOOK:
+      return state.filter((book) => book.id !== action.id);
+    default:
```

Comment:
- [ ] The project requirement requires you to define a slice, it is talking of the usage of createSlice method which `@reduxjs/toolkit` provides.
### Project requirement
![image](https://github.com/harshi0102/book-store/assets/86200356/ebbd1b4e-e829-4fb5-9026-1452067fc60f)


It would be something like the following:
```
import { createSlice } from "@reduxjs/toolkit"

export const booksSlice = createSlice({
  name: "books",
  initialState: {
    books: [],
  },
  reducers: {
    addBook: (state, action) => {
      state.books = [
        ...state.books,
        {
          id: Math.floor(Math.random() * 100),
          title: action.payload.title,
          author: action.payload.author,
        },
      ]
    },
    removeBook: (state, action) => {
      state.books = state.books.filter((book) => book.id !== action.id)
    },
  },
})

export const { addBook, removeBook } = booksSlice.actions

export default booksSlice.reducer
```

You can read more on it [here](https://redux-toolkit.js.org/api/createSlice)